highfidelity/hifi
# Case 20427: Avatar Exporter v0.3.5 - add scaling, fix colon prefixed joint names and material issues, add exporter version field

https://highfidelity.fogbugz.com/f/cases/20427/
https://highfidelity.fogbugz.com/f/cases/21685/
https://highfidelity.fogbugz.com/f/cases/21718/
https://highfidelity.fogbugz.com/f/cases/21682/

Changes on master: https://github.com/highfidelity/hifi/pull/15194

New Avatar Exporter package:
https://github.com/highfidelity/hifi/raw/b714b28809267d8459d87ee550aa81b063b02db3/tools/unity-avatar-exporter/avatarExporter.unitypackage

Pre-merge test plan:
- Download the following files:
https://github.com/highfidelity/hifi/raw/b714b28809267d8459d87ee550aa81b063b02db3/tools/unity-avatar-exporter/avatarExporter.unitypackage
https://hifi-content.s3.us-west-1.amazonaws.com/davidback/avatars/PR15174/Halo%203%20Master%20Chief%20%5BUpdated%5D.unitypackage
https://hifi-content.s3.us-west-1.amazonaws.com/davidback/avatars/PR15174/TrumpUnity5.unitypackage
- Using Unity version 2018.2.12f1 (install with https://download.unity3d.com/download_unity/0a46ddfcfad4/Windows64EditorInstaller/UnitySetup64-2018.2.12f1.exe if you don't have it) and open a new Unity project in it
- Open the above avatarExporter.unitypackage and import it into your Unity project
- Ensure the version is 0.3.5 in High Fidelity menu > About as well as in the included README file that was added to your Unity Assets panel
- Open the above .unitypackage files in explorer and import their contents into your Unity project
- With Halo 3 Master Chief, select the actual avatar model in your Unity assets and choose High Fidelity menu > Export New Avatar
- Verify a new preview scene opens in the Unity Scene view that contains the avatar in front of a wall with colored ranges for red (too tall/short), yellow (slightly short/tall), green (recommended height), as well as height markers on the wall for 0.5 - 2.5m
- Verify when the Export New Avatar window opens that Scale is automatically adjusted to 2.38% for you so that the avatar's height is within the green range, and that a message appears saying your scale had been automatically adjusted and disappears once you peform an action on the export window
- Export the avatar, open the resulting avatar.fst file, and verify "exporterVersion = 0.3.5" is at the top as well as a scale value of 0.0238
- Back in Unity, select Halo 3 Master Chief again, and High Fidelity menu > Update Existing Avatar
- Choose the fst file you exported above, change the scale from 2.38% back still within the green range, and export the updated fst
- Verify in the fst that the new scale value is the percent you set / 100
- With Trump, select the actual avatar model in your Unity assets and choose High Fidelity menu > Export New Avatar
- Verify there are warnings about materials with alternate Standard shaders as well as materials with unsupported shaders
- Export the avatar without any changes
- Select the trump_head_d material in the Assets/Trump/Materials folder and change the Albedo color to something very different
- Export the avatar again with a different project name
- Upload the above 3 avatar projects to S3 and ask me to upload them for you
- Verify all 3 avatars appear as expected and animated as expected in Interface which includes Master Chief being an appropriate height and 2nd Trump appearing with different colored head from the 1st Trump
